vulyk/blueprints/gamification/core/queries.py

```diff
@@ -59,14 +59,15 @@ def __init__(self, rule: Rule) -> None:
         """
         super().__init__()
 
+        self._filter_first['answer'] = {'$exists': True}
```

Comment:
Could you please create a dedicated test case to ensure that no single unfinished session would slip in?

Reply:
Done